### PR TITLE
Fixes #77 - Added new nameof() tests to verify the fix for GMB-6506

### LINKED ISSUE #77: BasicNameofTestSuite to test if nameof() returns proper values
### Description

As there seems to be bug in `name()` when using enums in 2024.6 (their values are returned instead of their name), I've added test which should check if that really occurs for selected runtime.

### Steps To The Task

- [x] add tests for nameof() for enums 
- [x] add tests for nameof() for global, macro and local variables

### PR DESCRIPTION
**Pull request by gnysek**
Adds tests for `nameof()` (issue #77).

This is a test which could be helpful after fixing https://github.com/YoYoGames/GameMaker-Bugs/issues/6506 .

### CODE REVIEW

**Comment by YYDan** [comment]
Please always file an issue describing the changes you want to add before adding a PR for undocumented changes.

**Comment by gnysek** [comment]
This is for #77

**Comment by YYDan** [comment]
Thanks - you beat me to it, but I was about to apologise and say I realised that the contribution guidelines in the Readme were _far_ too simplistic and so I have updated them now to be correct ;)

**Comment by gnysek** [comment]
Yeah, there was no info about it, that's why I didn't added issue in first place :) But no problem!

File: projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#macro NAMEOF_TEST_STRING "foobar"
+enum nameof_test {
+	entry_1,
+	entry_2,
+}
+global.NAMEOF_GLOBAL_TEST_REAL = 21.37;
+global.NAMEOF_GLOBAL_TEST_STRING = "foobar";
+
+function BasicNameofTestSuite() : TestSuite() constructor {
+
```
Comment (AdamBorek) [inline projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml]:
```suggestion

	// Test names of macros
```

File: projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+global.NAMEOF_GLOBAL_TEST_REAL = 21.37;
+global.NAMEOF_GLOBAL_TEST_STRING = "foobar";
+
+function BasicNameofTestSuite() : TestSuite() constructor {
+	
+	addFact("nameof_macro_test", function() {
+		assert_equals(nameof(NAMEOF_TEST_REAL), "NAMEOF_TEST_REAL", "#1 nameof of macro which holds real value");
+		assert_equals(nameof(NAMEOF_TEST_STRING), "NAMEOF_TEST_STRING", "#2 nameof of macro which holds string value");
+	});
+
```
Comment (AdamBorek) [inline projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:20]:
```suggestion
	
	// Test names of enums
```

File: projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	addFact("nameof_macro_test", function() {
+		assert_equals(nameof(NAMEOF_TEST_REAL), "NAMEOF_TEST_REAL", "#1 nameof of macro which holds real value");
+		assert_equals(nameof(NAMEOF_TEST_STRING), "NAMEOF_TEST_STRING", "#2 nameof of macro which holds string value");
+	});
+	
+	addFact("nameof_enum_test", function() {
+		assert_equals(nameof(nameof_test.entry_1), "nameof_test.entry_1", "#3 nameof of enum which holds real value");
+		assert_equals(nameof(nameof_test.entry_2), "nameof_test.entry_2", "#4 nameof of enum which holds string value");
+	});
+
```
Comment (AdamBorek) [inline projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:26]:
```suggestion
	
	// Test names of global variables
```

File: projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	addFact("nameof_enum_test", function() {
+		assert_equals(nameof(nameof_test.entry_1), "nameof_test.entry_1", "#3 nameof of enum which holds real value");
+		assert_equals(nameof(nameof_test.entry_2), "nameof_test.entry_2", "#4 nameof of enum which holds string value");
+	});
+	
+	addFact("nameof_global_test", function() {
+		assert_equals(nameof(global.NAMEOF_GLOBAL_TEST_REAL), "global.NAMEOF_GLOBAL_TEST_REAL", "#5 nameof of global which holds real value");
+		assert_equals(nameof(global.NAMEOF_GLOBAL_TEST_STRING), "global.NAMEOF_GLOBAL_TEST_STRING", "#6 nameof of global which holds string value");
+	});
+
```
Comment (AdamBorek) [inline projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:32]:
```suggestion
	
	// Test names of local variables
```

File: projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#macro NAMEOF_TEST_REAL 21.37
```
Comment (AdamBorek) [inline projects/xUnit/scripts/BasicNameofTestSuite/BasicNameofTestSuite.gml:2]:
```suggestion
// Add variables to test names of
#macro NAMEOF_TEST_REAL 21.37
```

**Review by AdamBorek — COMMENTED** [review]
All looks good, just added some comments above the facts and declarations

**Comment by gnysek** [comment]
I've added another test, as during checking Squirrel (`-I`) version of betas, I found an interesting bug, that if there's `\t` in front of Tab, macro would be converted to empty string in case of `nameof()`. It's then worth to check it too in future tests, to prevent regression.

**Comment by YYDan** [comment]
Thanks - although, if adding more tests later on, ensure you add a separate new issue and PR for just _that bit_ rather than adding it later on into an existing PR awaiting review ;)

**Review by DiasFranciscoA — APPROVED** [review]

**Comment by DiasFranciscoA** [comment]
@gnysek apparently there are some conflicts with the dest branch, please review this PR so we can merge it

**Comment by gnysek** [comment]
@DiasFranciscoA I'm not sure if `.resource_order` file should be in this repo. It could cause problems every time that during one pull request is made, another one with another new resource(s) is merged.
I've fixed it so merged file contains all resources, then reopened project in GM, to ensure .resource_order is up to date.

**Comment by DiasFranciscoA** [comment]
Hmm that makes sense, I'll make sure `.resource_order` is removed from the repo and added to `.gitignore`

**Comment by DiasFranciscoA** [comment]
@gnysek can you please resolve the conflicts the `.resource_order` file is now out of the picture.

EDITED: Fixed locally and pushed to develop (thank you for the contribution)

**Pull request closed without merging**